tests/conftest.py

```diff
@@ -238,3 +240,41 @@ def unprivileged_client(
 
         else:
             yield admin_client
+
+
+@pytest.fixture(scope="session")
+def dsc_resource(admin_client: DynamicClient):
+    name = py_config["dsc_name"]
+    try:
+        for dsc in DataScienceCluster.get(dyn_client=admin_client, name=name):
+            yield dsc
+    except Exception:
+        raise ResourceNotFoundError
```

Comment:
how about:

```suggestion
    name = py_config["dsc_name"]
    for dsc in DataScienceCluster.get(dyn_client=admin_client, name=name):
            return dsc

        raise ResourceNotFoundError(f"(could not find {name} DCS")
```
no need to yield here as there is not teardown